### PR TITLE
Quarto

### PR DESCRIPTION
**Pull request by asshah4**
Add Quarto markdown parser. This is similar to that of RMarkdown. Added unit tests that appear to be working on local build.

### CODE REVIEW

File: Tmain/output-encoding-option.d/tags-e [hunk 1]
```diff
@@ -0,0 +1,13 @@
+!_TAG_FILE_ENCODING	shift_jis	//
```
Comment (masatake) [inline Tmain/output-encoding-option.d/tags-e:1]: I don't understand why this file is needed.
Maybe you may want to make another pull request and this file may be part of it.

File: Tmain/input-encoding-option.d/tags-e [hunk 2]
```diff
@@ -0,0 +1,13 @@
+!_TAG_FILE_ENCODING	UTF-8	//
```
Comment (masatake) [inline Tmain/input-encoding-option.d/tags-e:1]: I don't understand why this file is needed.
Maybe you may want to make another pull request and this file may be part of it.

File: Tmain/list-subparsers-all.d/stdout-expected.txt [hunk 3]
```diff
@@ -18,6 +18,7 @@ PythonLoggingConfig  Iniconf    base <> sub {bidirectional}
 QtMoc                C++        base <> sub {bidirectional}
 R6Class              R          base <> sub {bidirectional}
 RMarkdown            Markdown   base <= sub {dedicated}
+Quarto 							 Markdown   base <= sub {dedicated}
```
Comment (masatake) [inline Tmain/list-subparsers-all.d/stdout-expected.txt:21]: make tmain may report a failure for this.

File: Units/parser-quarto.r/frontmatter.d/input.qmd [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+title: example for u-ctags
```
Comment (masatake) [inline Units/parser-quarto.r/frontmatter.d/input.qmd:2]: *I* will find the way to extract this title. (@masatake's task).

File: man/ctags-json-output.5 [hunk 5]
```diff
@@ -0,0 +1,169 @@
+.\" Man page generated from reStructuredText.
```
Comment (masatake) [inline man/ctags-json-output.5:1]: Don't put the .5 file to our git repo.

File: misc/git-tag-maybe.sh [hunk 6]
```diff
@@ -22,7 +22,7 @@ run() {
 
 base=6.0
 cal=$(date +%Y%m%d)
-chicken=0
```
Comment (masatake) [inline misc/git-tag-maybe.sh:25]: Don't edit this.

File: packcc.dSYM/Contents/Info.plist [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (masatake) [inline packcc.dSYM/Contents/Info.plist:1]: Why don't you do this?

File: win32/config_mingw.h [hunk 8]
```diff
@@ -506,7 +506,7 @@
 #define PACKAGE_NAME "universal-ctags"
 
 /* Define to the full name and version of this package. */
-#define PACKAGE_STRING "universal-ctags 6.0.0"
+#define PACKAGE_STRING "universal-ctags 6.0.1"
```
Comment (masatake) [inline win32/config_mingw.h:509]: Don't edit this for now.

File: win32/config_mingw.h [hunk 9]
```diff
@@ -515,7 +515,7 @@
 #define PACKAGE_URL ""
 
 /* Define to the version of this package. */
-#define PACKAGE_VERSION "6.0.0"
+#define PACKAGE_VERSION "6.0.1"
```
Comment (masatake) [inline win32/config_mingw.h:518]: Don't edit this for now.

File: win32/ctags.exe.manifest [hunk 10]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1">
-  <assemblyIdentity type="win32" name="Universal Ctags" version="6.0.0.0" />
+  <assemblyIdentity type="win32" name="Universal Ctags" version="6.0.1.0" />
```
Comment (masatake) [inline win32/ctags.exe.manifest:3]: Ditto.

File: win32/ctags.rc [hunk 11]
```diff
@@ -76,12 +76,12 @@ BEGIN
         BEGIN
             VALUE "CompanyName", "Universal Ctags Team"
             VALUE "FileDescription", "Universal Ctags"
-            VALUE "FileVersion", "6.0.0.0"
+            VALUE "FileVersion", "6.0.1.0"
```
Comment (masatake) [inline win32/ctags.rc:79]: Ditto.

File: win32/ctags.rc [hunk 12]
```diff
@@ -76,12 +76,12 @@ BEGIN
...
             VALUE "InternalName", "ctags.exe"
             VALUE "LegalCopyright", "Copyright \251 2015-2022 Universal Ctags Team"
             VALUE "OriginalFilename", "ctags.exe"
             VALUE "ProductName", "Universal Ctags"
-            VALUE "ProductVersion", "6.0.0.0"
+            VALUE "ProductVersion", "6.0.1.0"
```
Comment (masatake) [inline win32/ctags.rc:84]: Ditto.

File: win32/ctags_vs2013.vcxproj [hunk 13]
```diff
@@ -472,4 +473,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
\ No newline at end of file
+</Project>
```
Comment (masatake) [inline win32/ctags_vs2013.vcxproj:476]: Don't insert a newline here.

File: win32/ctags_vs2013.vcxproj.filters [hunk 14]
```diff
@@ -920,4 +923,4 @@
       <Filter>Resource Files</Filter>
     </ResourceCompile>
   </ItemGroup>
-</Project>
```
Comment (masatake) [inline win32/ctags_vs2013.vcxproj.filters:923]: Don't insert newline here.

File: docs/man/ctags-lang-kconfig.7.rst [hunk 15]
```diff
@@ -0,0 +1,35 @@
+.. _ctags-lang-kconfig(7):
```
Comment (masatake) [inline docs/man/ctags-lang-kconfig.7.rst:1]: Don't you add this in your pull request?

File: configure.ac [hunk 16]
```diff
@@ -7,7 +7,7 @@
 #	Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.64])
-AC_INIT([universal-ctags],[6.0.0])
+AC_INIT([universal-ctags],[6.0.1])
```
Comment (masatake) [inline configure.ac:10]: Don't edit this.

File: parsers/quarto.c [hunk 17]
```diff
@@ -0,0 +1,135 @@
+/*
+ *
+ *  Copyright (c) 2022, Masatake YAMATO
```
Comment (masatake) [inline parsers/quarto.c:3]: As far as quickly reading, this parser is mostly the same as Rmarkdown parser.
Do we need a new parser for Quarto? If yes, why do you think so?

**Comment by masatake** [comment]
> Add Quarto markdown parser. This is similar to that of RMarkdown. Add unit tests but

but..what?

**Comment by k-takata** [comment]
It looks that `packcc.dSYM/Contents/Resources/DWARF/packcc` is mistakenly added. Please remove it.

File: configure.ac (hunk 16 above)
Comment (asshah4) [reply configure.ac:10]: I apologize - I attempted to follow the online recommendations, but makes more sense for that to be only for the integral editors/authors.

File: man/ctags-json-output.5 (hunk 5 above)
Comment (asshah4) [reply man/ctags-json-output.5:1]: Removed this file in next commit

File: parsers/quarto.c (hunk 17 above)
Comment (asshah4) [reply parsers/quarto.c:3]: I think we do for the following reasons:

1. The QMD quarto "chunks" use slightly different pattern for labeling that RMD files do which may affect how tags are extracted
1. Although similar, to be able to use tag-bar style plugins (e.g. as in vim), would need the qmd extension to be supported.
1. The QMD style seems to still be evolving with varying language support, so likely may start diverging more and more from RMD

As a heavy RMD user that has switched to Quarto recently, I personally would enjoy being able to easily extract tags with minimal increased overhead (essentially took the __Rmarkdown__ code and revised it slightly)

**Comment by masatake** [comment]
Could you reorganize your pull request?
I don't want to review commits removing something you added in another commit.

**Comment by masatake** [comment]
This includes try-and-error commits. 
See https://docs.ctags.io/en/latest/contributions.html#squashing-commits.

I wanted to see the Quarto input that the RMarkdown parser doesn't handle well.
In other words, the input illustrating why the Quarto parser is needed was not shown.

**Comment by asshah4** [comment]
Thanks @masatake - I'll reorganize the pull-request and show a better example about what the error may be.

**Comment by masatake** [comment]
Why don't you show any example that your Quarto parser accepts but RMarkdown doesn't accept? Though it is a small but certain starting point of our game.
I guess Quarto is mostly the same as RMarkdown. The difference may be so small, so we don't need a new parser.

**Comment by masatake** [comment]
Those who are interested in supporting Quarto in ctags, see #3677.
